sphinx_search/extension.py

```diff
@@ -24,21 +30,39 @@ def copy_asset_files(app, exception):
 
 
 def inject_static_files(app):
-    """Inject correct CSS and JS files based on the value of ``RTD_SPHINX_SEARCH_FILE_TYPE``."""
-    file_type = app.config.rtd_sphinx_search_file_type
-    expected_file_type = ASSETS_FILES.keys()
+    """
+    Inject correct CSS and JS files based on the value of ``RTD_SPHINX_SEARCH_FILE_TYPE``.
+
+    This only injects file if the docs are build on Read the Docs.
+    """
+
+    # This is done to properly test the extension.
+    # If the tests are being executed, add `html`
+    # as one of the `ONLINE_BUILDERS`, so that
+    # proper CSS and JS files gets injected and can be tested.
+    if os.environ.get('TESTING_ENVIRONMENT') == 'True':
+        ONLINE_BUILDERS.append('html')
+
+    on_rtd = os.environ.get('READTHEDOCS') == 'True'
 
-    assert (
-        file_type in expected_file_type
-    ), f'"{file_type}" file type is not supported.'
+    # only inject files if the builder is one of the ONLINE_BUILDERS
+    # and on_rtd is True
+    if app.builder.name in ONLINE_BUILDERS and on_rtd:
 
-    files = ASSETS_FILES[file_type]
+        file_type = app.config.rtd_sphinx_search_file_type
+        expected_file_type = ASSETS_FILES.keys()
 
-    for file in files:
-        if file.endswith('.js'):
-            app.add_js_file(file)
-        elif file.endswith('.css'):
-            app.add_css_file(file)
+        assert (
+            file_type in expected_file_type
+        ), f'"{file_type}" file type is not supported.'
```

Comment:
What kind of UX does this show the user? I'm guessing we should use proper Sphinx warnings and exceptions instead of raw assets.